mercantile/__init__.py

```diff
@@ -133,7 +133,20 @@ def _parse_tile_arg(*args):
     if len(args) == 1:
         args = args[0]
     if len(args) == 3:
-        return Tile(*args)
+        tile = Tile(*args)
+
+        xtile, ytile, zoom = tile
+
+        if zoom != int(zoom) or not int(zoom) >= 0:
+            raise TileArgParsingError("zoom must be a positive integer")
+
+        if xtile != int(xtile) or not (0 <= int(xtile) <= (2 ** zoom - 1)):
+            raise TileArgParsingError("x must be an integer in [0, 2 ** z - 1]")
+
+        if ytile != int(ytile) or not (0 <= int(ytile) <= (2 ** zoom - 1)):
+            raise TileArgParsingError("y must be an integer in [0, 2 ** z - 1]")
```

Comment:
Web mercator breaks down beyond the Y limits for a zoom level, right? `y < 0` and `y >= 2 ** z` can't exist. So I'm in favor of validating here.

For X, though... `x < 0` and `x >= 2 ** z` aren't mathematically impossible. We could let this case go, deferring it to future work on making `Tile` more strict. What would you think about that?

Reply:
Hm. If we validate y, will it break the API? If so we should also validate x for consistency reasons, or no?

Reply:
After more thought, it's clear to me that validating Y does break the API. The line of code I reference at https://github.com/mapbox/mercantile/pull/125#discussion_r610109267 would be broken by introducing validation. The project can work around that, but there's a non-zero probability that users adapted that code and depend on Y not being validated.

So, let's remove the checks for now. In a follow up PR I will add some warnings about changes we may make in 2.0.

Reply:
:+1: